homeassistant/components/iotawatt/coordinator.py

```diff
@@ -32,6 +32,24 @@ def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
             update_interval=timedelta(seconds=30),
         )
 
+        self._last_run = None
+        self._refresh_requested = False
+
+    def update_last_run(self, last_run: datetime):
+        """Notify coordinator of a sensor last update time."""
+        # We want to fetch the data from the iotawatt since HA was last shutdown.
+        # We retrieve from the sensor last updated.
+        # This method is called from each sensor upon their state being restored.
+        if self._last_run is None or last_run > self._last_run:
+            self._last_run = last_run  # type: ignore
+
+    async def request_refresh(self):
+        """Request a refresh of the iotawatt sensors."""
+        if self._refresh_requested:
+            return
+        self._refresh_requested = True
+        await self.async_request_refresh()
+
```

Comment:
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/update_coordinator.py#L141

`async_request_refresh` already has a debounce to prevent multiple refreshes in the window.

Reply:
Good to know. I'll remove the unnecessary safeguards then